src/registrar/views/domain.py

```diff
@@ -138,10 +138,17 @@ def get_initial(self):
         """The initial value for the form (which is a formset here)."""
         domain = self.get_object()
         nameservers = domain.nameservers
-        if nameservers is None:
-            return []
+        initial_data = []
 
-        return [{"server": name} for name, *ip in domain.nameservers]
+        if nameservers is not None:
+            # Add existing nameservers as initial data
+            initial_data.extend({"server": name} for name, *ip in nameservers)
+
+        # Ensure at least 3 fields, filled or empty
+        while len(initial_data) < 2:
```

Comment:
Should this be `<= 2` or `< 3` according to the comment?

Reply:
Nope, cuz there's a +1 formset in the form definition. Good eye though!